Android lint tests

ciera opened the pull request.

supertriceratops commented:
Assigning to collinw in case you want more feedback before Friday.

collinw commented on shipshape/androidlint_analyzer/androidlint/analyzer.go:
```diff
@@ -51,7 +51,7 @@ func (Analyzer) Category() string { return "AndroidLint" }
 // referenced by the list of files on the depot path. In case of an error, it
 // returns partial results.
-func (ala *Analyzer) Analyze(ctx *ctxpb.ShipshapeContext) ([]*notepb.Note, error) {
+func (ala Analyzer) Analyze(ctx *ctxpb.ShipshapeContext) ([]*notepb.Note, error) {
```
Why does the type of the invocant change?

ciera replied:
It was needed to call within the tests. It can still be called with a reference, but now you can call with either a reference or an actual object.

collinw commented:
I'm happy to have these files live in their own testdata directory; that makes it clear what the scope of their usage is.

Overall LGTM.

supertriceratops commented:
I meant, are there any licensing issue with copying these files into our testdata directory? Sounds like ready to update the README.md file and merge.

ciera commented:
I found a more recent version and am updating the branch to use that instead. The code Apache licensed and from github...should this be under third_party then?

collinw commented:
As long as the licensing is clearly indicated via a README or LICENSE file, I think pragmatism (keeping the tests close to the code-under-test) can make the case for not needing it in third_party.

ciera commented:
PTAL, I addressed all comments and added AndroidLint to the end-to-end test as well.

I'm having git trouble; I'm trying to understand how to properly rebase here.

ciera commented:
Ok, this is ready now. Answer is: once a pull request has gone through, don't amend the commits. If you do, you have to force the push.

collinw commented:
LGTM

supertriceratops commented:
LGTM
